### DISCUSSION TITLE
Newbie linux install straight from github.

### DISCUSSION THREAD

**Discussion by petersilva** [opendcs/opendcs#251] [General]
I'm not sure that there is anything wrong here, just sharing what happenned... maybe useful to someone? maybe not.

My goal, to calibrate expectations:

* using a  ubuntu-22.04 VM (headless no gui)
* to run getDcpMessages to poll USGS servers for some Canadian surface AWOS data.
* that will be chopped into 1 file/ob, and then turned into WMO-GTS style messages and forwarded using our currently preferred methods.
* command line is easiest in this environment.

This is to modernize a twenty year old working example for an old telecom system, and I'm making new integration
for our current telecom stuff.  there are about 5000 stations listed in the DB, and we poll 24x7x365 (Canadian weather service.)

I started reading here:

https://opendcs-env.readthedocs.io/en/latest/install-guide.html#single-user-opendcs-installation

It says to java -jar "the file downloaded from github" ... taking a guess, I fetched the latest release.

```
wget https://github.com/opendcs/opendcs/releases/download/7.0.8/opendcs-7.0.8.jar
ubuntu@lrgs:~$ java -jar opendcs-7.0.8.jar
no main manifest attribute, in opendcs-7.0.8.jar
ubuntu@lrgs:~$

```

hmm... OK so I saw there was another -ot- file? ...

```
 wget https://github.com/opendcs/opendcs/releases/download/7.0.8/opendcs-ot-7.0.8.jar
 java -jar opendcs-ot-7.0.8.jar
```

So I ran that and it gave me text prompts, and that seemed to go OK.

What is ot? is that ok? anyways. I set up the environment with DCSTOOL_HOME=/home/ubuntu/opendcs
and tried to start the next bit:

```

ubuntu@lrgs:~$ launcher_start
Exception in thread "main" java.awt.HeadlessException: 
No X11 DISPLAY variable was set, but this program performed an operation which requires it.
	at java.desktop/java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:208)
	at java.desktop/java.awt.Window.<init>(Window.java:548)
	at java.desktop/java.awt.Frame.<init>(Frame.java:423)
	at java.desktop/java.awt.Frame.<init>(Frame.java:388)
	at java.desktop/javax.swing.JFrame.<init>(JFrame.java:180)
	at decodes.launcher.LauncherFrame.<init>(LauncherFrame.java:258)
	at decodes.launcher.LauncherFrame.main(LauncherFrame.java:1231)
ubuntu@lrgs:~$

```

So Linux in general is going through a transition of display managers ... from X11 to Wayland.  Ubuntu 22.04 is Wayland by default.
Is there a way to just not use a GUI? What is this set up for? 

Anyways... I figured out that I don't actually need to do anything more. getDcpMessages already works to query USGS when I tried it. so... I'm confused, but it's doing what I need.

Thanks!

**Comment by MikeNeilson** [comment]
FYI for LRGS everything can be configured manually in text files. the main one is lrgs.conf, then there's ddsrecv and drgsrecv xml files. I think we have some examples in the document.

Otherwise ssh with X forwarding and if you're on windows X windows from cygwin or something similar.

Making that all easier is on the todo list, just going to be awhile unfortunately.


If you want to use the DECODES engine to actually handle the processing you can use the XML database and do the configuration on a local machine with a gui and push the xml database files to the host. Not ideal, but it's a pretty common usage.

**Reply by petersilva** [reply]
I guess what would be great for me is pointers in the installation manual to:
* which configuration files exist (which ones are needed ffor doing what.)
* the syntax ( I think this is already there.) 

It's really just the first one.  the installation guide provides a GUI walk through, but without the GUI, I don't know which config file I have to tweak.

I will be looking into decode in a while ... I have some code that "splits" the output of getDcpMessages into files, but it's archeology for now.  not sure if using the code we have or using LRGS built-in decode stuff is the better path.  Will likely end up with a mix of both. I remember there were some Canadian stations that don't have correct decodes in LRGS (more accurate to say they didn't 20 years ago when this was set up. it could well have been addressed in the mean-time... no way to know really.) We had our own downstream decodes.

fwiw... I mentioned... in current linux distributions, there is no X I don't mean no screen, I mean "X" is being replaced by Wayland... so there is no x server to connect to.  yes, there is probably a way to set up a nested x server in wayland, but I certainly have never used it, and nothing else requires it.  so... sometime soon, the approach of using X from a Java App is going to be a problem.

**Reply by MikeNeilson** [reply]
I see. For the config files they don't all have extensions, unfortuantely, but they are either java properties files or XML files. what's in the directory of the default install is what's available... and I'm pretty sure nothing is missing.


As of the X vs Wayland issue. It's all Java Swing or AWT. In theory that should take care of that all by itself, OpenDCS isn't directly connecting to X11 just using the Java GUI bits. But I'll look into that in case I'm wrong. I think all my Ubuntu installs still have Xorg running.
